Make the repo working with the latest version of manif

Required after https://github.com/artivis/manif/pull/190


```
home/gromualdi/robot-code/robotology-superbuild/src/bipedal-locomotion-framework/src/TSID/src/SE3Task.cpp: In member function ‘bool BipedalLocomotion::TSID::SE3Task::setReferenceTrajectory(const SE3d&, const Tangent&, const Tangent&)’:
/home/gromualdi/robot-code/robotology-superbuild/src/bipedal-locomotion-framework/src/TSID/src/SE3Task.cpp:149:83: error: ‘const Tangent’ {aka ‘const struct manif::SE3Tangent<double>’} has no member named ‘v’
  149 |     ok = ok && m_R3Controller.setDesiredState({I_H_F.translation(), mixedVelocity.v()});
      |                                                                                   ^
/home/gromualdi/robot-code/robotology-superbuild/src/bipedal-locomotion-framework/src/TSID/src/SE3Task.cpp:149:87: error: cannot convert ‘<brace-enclosed initializer list>’ to ‘const State&’ {aka ‘const std::tuple<manif::Rn<double, 3>, manif::RnTangent<double, 3> >&’}
  149 |     ok = ok && m_R3Controller.setDesiredState({I_H_F.translation(), mixedVelocity.v()});
      |                                                                                       ^
In file included from /home/gromualdi/robot-code/robotology-superbuild/build/install/include/LieGroupControllers/impl/ProportionalDerivativeController/Controller.h:15,
                 from /home/gromualdi/robot-code/robotology-superbuild/build/install/include/LieGroupControllers/ProportionalDerivativeController.h:11,
                 from /home/gromualdi/robot-code/robotology-superbuild/src/bipedal-locomotion-framework/src/TSID/include/BipedalLocomotion/TSID/SE3Task.h:15,
                 from /home/gromualdi/robot-code/robotology-superbuild/src/bipedal-locomotion-framework/src/TSID/src/SE3Task.cpp:8:
/home/gromualdi/robot-code/robotology-superbuild/build/install/include/LieGroupControllers/impl/ProportionalDerivativeController/ControllerBase.h:87:83: note:   initializing argument 1 of ‘bool LieGroupControllers::ProportionalDerivativeControllerBase<_Derived>::setDesiredState(const State&) [with _Derived = LieGroupControllers::ProportionalDerivativeController<manif::Rn<double, 3> >; LieGroupControllers::ProportionalDerivativeControllerBase<_Derived>::State = std::tuple<manif::Rn<double, 3>, manif::RnTangent<double, 3> >]’
   87 | bool ProportionalDerivativeControllerBase<_Derived>::setDesiredState(const State& state)
      |                                                                      ~~~~~~~~~~~~~^~~~~
/home/gromualdi/robot-code/robotology-superbuild/src/bipedal-locomotion-framework/src/TSID/src/SE3Task.cpp:150:64: error: ‘const Tangent’ {aka ‘const struct manif::SE3Tangent<double>’} has no member named ‘v’
  150 |     ok = ok && m_R3Controller.setFeedForward(mixedAcceleration.v());
      |                                                                ^
/home/gromualdi/robot-code/robotology-superbuild/src/bipedal-locomotion-framework/src/TSID/src/SE3Task.cpp:152:77: error: ‘const Tangent’ {aka ‘const struct manif::SE3Tangent<double>’} has no member named ‘w’
  152 |     ok = ok && m_SO3Controller.setDesiredState({I_H_F.quat(), mixedVelocity.w()});
      |                                                                             ^
/home/gromualdi/robot-code/robotology-superbuild/src/bipedal-locomotion-framework/src/TSID/src/SE3Task.cpp:152:81: error: cannot convert ‘<brace-enclosed initializer list>’ to ‘const State&’ {aka ‘const std::tuple<manif::SO3<double>, manif::SO3Tangent<double> >&’}
  152 |     ok = ok && m_SO3Controller.setDesiredState({I_H_F.quat(), mixedVelocity.w()});
      |                                                                                 ^
In file included from /home/gromualdi/robot-code/robotology-superbuild/build/install/include/LieGroupControllers/impl/ProportionalDerivativeController/Controller.h:15,
                 from /home/gromualdi/robot-code/robotology-superbuild/build/install/include/LieGroupControllers/ProportionalDerivativeController.h:11,
                 from /home/gromualdi/robot-code/robotology-superbuild/src/bipedal-locomotion-framework/src/TSID/include/BipedalLocomotion/TSID/SE3Task.h:15,
                 from /home/gromualdi/robot-code/robotology-superbuild/src/bipedal-locomotion-framework/src/TSID/src/SE3Task.cpp:8:
/home/gromualdi/robot-code/robotology-superbuild/build/install/include/LieGroupControllers/impl/ProportionalDerivativeController/ControllerBase.h:87:83: note:   initializing argument 1 of ‘bool LieGroupControllers::ProportionalDerivativeControllerBase<_Derived>::setDesiredState(const State&) [with _Derived = LieGroupControllers::ProportionalDerivativeController<manif::SO3<double> >; LieGroupControllers::ProportionalDerivativeControllerBase<_Derived>::State = std::tuple<manif::SO3<double>, manif::SO3Tangent<double> >]’
   87 | bool ProportionalDerivativeControllerBase<_Derived>::setDesiredState(const State& state)
      |                                                                      ~~~~~~~~~~~~~^~~~~
/home/gromualdi/robot-code/robotology-superbuild/src/bipedal-locomotion-framework/src/TSID/src/SE3Task.cpp:153:65: error: ‘const Tangent’ {aka ‘const struct manif::SE3Tangent<double>’} has no member named ‘w’
  153 |     ok = ok && m_SO3Controller.setFeedForward(mixedAcceleration.w());
      |                                                                 ^
make[2]: *** [src/TSID/CMakeFiles/TSID.dir/build.make:76: src/TSID/CMakeFiles/TSID.dir/src/SE3Task.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:3077: src/TSID/CMakeFiles/TSID.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -17,7 +17,7 @@ env:
   Catch2_TAG: v2.13.4
   Qhull_TAG: 2020.2
   CasADi_TAG: 3.5.5.2
-  manif_TAG: 0.0.3
+  manif_TAG: 85cad9841b0ac4800e10e38c634c397d401e6375
   matioCpp_TAG: v0.1.0
   LieGroupControllers_TAG: v0.0.1
   osqp_TAG: v0.6.2

--- a/.github/workflows/conda-forge-ci.yml
+++ b/.github/workflows/conda-forge-ci.yml
@@ -8,6 +8,9 @@ on:
   # Execute a "nightly" build at 2 AM UTC
   - cron:  '0 2 * * *'
 
+env:
+  manif_TAG: 85cad9841b0ac4800e10e38c634c397d401e6375
+
 jobs:
   build:
     name: '[${{ matrix.os }}@${{ matrix.build_type }}@conda]'
@@ -33,7 +36,37 @@ jobs:
         # Compilation related dependencies
         mamba install cmake compilers make ninja pkg-config
         # Actual dependencies
-        mamba install -c robotology-staging idyntree yarp matio-cpp lie-group-controllers eigen qhull casadi cppad manif spdlog catch2
+        mamba install -c robotology-staging idyntree yarp libmatio matio-cpp lie-group-controllers eigen qhull casadi cppad spdlog catch2
+
+    - name: Dependencies [manif - Linux&macOS]
+      if: contains(matrix.os, 'macos') || contains(matrix.os, 'ubuntu')
+      shell: bash -l {0}
+      run: |
+        # manif
+        cd ${GITHUB_WORKSPACE}
+        git clone https://github.com/artivis/manif.git
+        cd manif
+        git checkout ${manif_TAG}
+        mkdir build
+        cd build
+        cmake -GNinja -DCMAKE_INSTALL_PREFIX=${CONDA_PREFIX} ..
+        cmake --build . --config ${{ matrix.build_type }} --target install
+
+    - name: Dependencies [manif - windows]
+      if: contains(matrix.os, 'windows')
+      shell: bash -l {0}
+      run: |
+        # manif
+        cd ${GITHUB_WORKSPACE}
+        git clone https://github.com/artivis/manif.git
+        cd manif
+        git checkout ${manif_TAG}
+        mkdir build
+        cd build
+        cmake -G"Visual Studio 16 2019" \
+              -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} \
+              -DCMAKE_INSTALL_PREFIX=${CONDA_PREFIX}/Library ..
+        cmake --build . --config ${{ matrix.build_type }} --target install
 
     - name: Configure [Linux&macOS]
       if: contains(matrix.os, 'macos') || contains(matrix.os, 'ubuntu')

--- a/src/IK/src/SE3Task.cpp
+++ b/src/IK/src/SE3Task.cpp
@@ -175,10 +175,10 @@ bool SE3Task::setSetPoint(const manif::SE3d& I_H_F, const manif::SE3d::Tangent& 
 
     bool ok = true;
     ok = ok && m_R3Controller.setDesiredState(I_H_F.translation());
-    ok = ok && m_R3Controller.setFeedForward(mixedVelocity.v());
+    ok = ok && m_R3Controller.setFeedForward(mixedVelocity.lin());
 
     ok = ok && m_SO3Controller.setDesiredState(I_H_F.quat());
-    ok = ok && m_SO3Controller.setFeedForward(mixedVelocity.w());
+    ok = ok && m_SO3Controller.setFeedForward(mixedVelocity.ang());
 
     return ok;
 }

--- a/src/IK/tests/SE3TaskTest.cpp
+++ b/src/IK/tests/SE3TaskTest.cpp
@@ -117,8 +117,8 @@ TEST_CASE("SE3 Task")
             SO3Controller.setGains(kp);
             R3Controller.setGains(kp);
 
-            SO3Controller.setFeedForward(desiredVelocity.w());
-            R3Controller.setFeedForward(desiredVelocity.v());
+            SO3Controller.setFeedForward(desiredVelocity.ang());
+            R3Controller.setFeedForward(desiredVelocity.lin());
             SO3Controller.setDesiredState(desiredPose.quat());
             R3Controller.setDesiredState(desiredPose.translation());
 

--- a/src/TSID/src/SE3Task.cpp
+++ b/src/TSID/src/SE3Task.cpp
@@ -146,11 +146,11 @@ bool SE3Task::setReferenceTrajectory(const manif::SE3d& I_H_F,
 {
 
     bool ok = true;
-    ok = ok && m_R3Controller.setDesiredState({I_H_F.translation(), mixedVelocity.v()});
-    ok = ok && m_R3Controller.setFeedForward(mixedAcceleration.v());
+    ok = ok && m_R3Controller.setDesiredState({I_H_F.translation(), mixedVelocity.lin()});
+    ok = ok && m_R3Controller.setFeedForward(mixedAcceleration.lin());
 
-    ok = ok && m_SO3Controller.setDesiredState({I_H_F.quat(), mixedVelocity.w()});
-    ok = ok && m_SO3Controller.setFeedForward(mixedAcceleration.w());
+    ok = ok && m_SO3Controller.setDesiredState({I_H_F.quat(), mixedVelocity.ang()});
+    ok = ok && m_SO3Controller.setFeedForward(mixedAcceleration.ang());
 
     return ok;
 }

--- a/src/TSID/tests/SE3TaskTest.cpp
+++ b/src/TSID/tests/SE3TaskTest.cpp
@@ -120,10 +120,10 @@ TEST_CASE("SE3 Task")
             SO3Controller.setGains({kp, kd});
             R3Controller.setGains({kp, kd});
 
-            SO3Controller.setFeedForward(desiredAcceleration.w());
-            R3Controller.setFeedForward(desiredAcceleration.v());
-            SO3Controller.setDesiredState({desiredPose.quat(), desiredVelocity.w()});
-            R3Controller.setDesiredState({desiredPose.translation(), desiredVelocity.v()});
+            SO3Controller.setFeedForward(desiredAcceleration.ang());
+            R3Controller.setFeedForward(desiredAcceleration.lin());
+            SO3Controller.setDesiredState({desiredPose.quat(), desiredVelocity.ang()});
+            R3Controller.setDesiredState({desiredPose.translation(), desiredVelocity.lin()});
 
             SO3Controller.setState({BipedalLocomotion::Conversions::toManifRot(
                         kinDyn->getWorldTransform(controlledFrame).getRotation()),

